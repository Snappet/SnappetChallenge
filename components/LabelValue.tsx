import React from 'react';
import { Grid, GridProps, Typography } from '@material-ui/core';

interface LabelValueProps extends GridProps {
  label: React.ReactNode;
  value: React.ReactNode;
}

const LabelValue: React.FC<LabelValueProps> = ({ label, value, ...props }) => {
  return (
    <Grid container item spacing={1} alignItems="baseline" {...props}>
      <Grid item>
        <Typography color="textSecondary">{label}:</Typography>
      </Grid>
      <Grid item>
        <Typography variant="body2">{value}</Typography>
      </Grid>
    </Grid>
  );
};

export default LabelValue;
