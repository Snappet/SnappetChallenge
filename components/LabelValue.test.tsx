import React from 'react';
import { render } from '@testing-library/react';

import LabelValue from './LabelValue';

describe('LabelValue', () => {
  it('matches snapshot', () => {
    const { container } = render(<LabelValue label="test" value="value" />);
    expect(container).toMatchSnapshot();
  });
});
