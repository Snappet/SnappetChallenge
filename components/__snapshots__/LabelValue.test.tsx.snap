// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelValue matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        test
        :
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        value
      </p>
    </div>
  </div>
</div>
`;
